### PR TITLE
fix z-issues on move to tab menu

### PR DESCRIPTION
**Pull request by npretto**
### Description

This fixes two issues with zIndex related to the "move to tab" menu
First issue is a zindex issue with grid maps: https://metaboat.slack.com/archives/C057T1QTB3L/p1698169540830189
move-to-tab-z-index

Second issue is a zindex issue when editing a link card: https://github.com/metabase/metabase/pull/34961/files#r1371364752

image

## Cause
Both issues were cause by me avoiding the portal for the menu for the first implementation.

I didn't use the portal because to avoid this:

https://github.com/metabase/metabase/assets/1914270/d0443929-4ea1-45bc-ac97-e3b17eba918a

Not using the portal made the menu a child of the dashcard actions panel, which is a child of the card, and so hovering the menu kept the the dashcard actions panel open. (Hovering the card, or having focus, keeps the actions panel visible)

That solution caused the two bugs above, I tried to solve them with via zIndex and no JS, but every thing that I fixed caused another zIndex issue so I stepped back and used the portal.

## Solution
In this PR I add a piece of state to check if there is an open sub menu, and in that case I keep the actions panel visible. 

I don't like that the state is currently in Dashcard, we could refactor `renderDashCardActions` to be a component so that we can keep the state internal to it.

### How to verify

1. Create/go to a dashboard with more than 1 tab
2. Create a link card, focus the input, the popover should open, hover the move to tab icon in the actions panel

For the second issue:
1. Create/go to a dashboard with more than 1 tab
2. Add a card with a map-> grid visualization
3. Hover the move to tab icon in the actions panel

### Demo

https://github.com/metabase/metabase/assets/1914270/fe965068-1b51-4b1e-bf14-bf0b2a84d483

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by npretto** [comment]
Current dependencies on/for this PR:
* master
  * **PR #35240** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35240" target="_blank">Graphite</a> 
    * **PR #35324** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35324" target="_blank">Graphite</a>  👈
    * **PR #35309** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35309" target="_blank">Graphite</a> 
      * **PR #35310** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35310" target="_blank">Graphite</a> 
    * **PR #35243** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35243" target="_blank">Graphite</a> 
    * **PR #35241** <a href="https://app.graphite.dev/github/pr/metabase/metabase/35241" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/35324?utm_source=stack-comment).

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx [hunk 1]
```diff
@@ -116,6 +116,9 @@ function DashCard({
   const [isPreviewingCard, setIsPreviewingCard] = useState(false);
   const cardRootRef = useRef<HTMLDivElement>(null);
 
+  const [isActionPanelSubmenuOpen, setIsActionPanelSubmenuOpen] =
```
Comment (npretto) [inline frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx:119]: This is the state mentioned in the PR body, that I'd like to move to a new component that replaces `renderDashCardActions`

**Review by kamilmielnik — APPROVED** [review]
I confirm both issues are fixed :+1:

**Comment by uladzimirdev** [comment]
z-index is problematic 100% and I had several PRs for similar problems. @npretto what kind of "another broken thing" did you have?

**Comment by npretto** [comment]
> z-index is problematic 100% and I had several PRs for similar problems.
It is 😅 especially without portals

> @npretto what kind of "another broken thing" did you have?
When I was using the portal it was full of issues like hovered(or focused) cards being on top of open menus for other cards

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ export const DashCardRoot = styled.div<DashCardRootProps>`
     shouldForceHiddenBackground && hiddenBackgroundStyle}
 `;
 
-export const DashboardCardActionsPanel = styled.div`
+export const DashboardCardActionsPanel = styled.div<{ isSubMenuOpen: boolean }>`
```
[the remark this answers is not in the document]
Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx:44]: That's interesting, I'm trying to think of a way to fix it but nothing comes to my mind

I think this small issue is the price to pay for not using portals (and not having all the other bugs)

@cdeweyx do you think can leave this small thing there? It should only happen when people focus a card and try to move another one

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx (hunk 1 above)
Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx:119]: I'll try to draft a stacked pr on top of this

**Review by ranquild — COMMENTED** [review]
Please address comments from @kamilmielnik

**Pull request merged**